src/NumericalAlgorithms/Spectral/Projection.cpp

```diff
@@ -0,0 +1,320 @@
+          CacheRange<2, maximum_number_of_points<Basis::Legendre> + 1>>
+          cache([](const size_t encoded_quadrature_element,
+                   const size_t extents_element,
+                   const size_t encoded_quadrature_mortar,
+                   const size_t extents_mortar) noexcept {
+            if (extents_element > extents_mortar) {
+              return Matrix{};
+            }
+            const Mesh<1> mesh_element(
+                extents_element, Basis::Legendre,
+                decode_quadrature(encoded_quadrature_element));
+            const Mesh<1> mesh_mortar(
+                extents_mortar, Basis::Legendre,
+                decode_quadrature(encoded_quadrature_mortar));
+
+            // The projection in spectral space is just a truncation
+            // of the modes.
+            const auto& spectral_to_grid_element =
+                spectral_to_grid_points_matrix(mesh_element);
+            const auto& grid_to_spectral_mortar =
+                grid_points_to_spectral_matrix(mesh_mortar);
+            Matrix projection(extents_element, extents_mortar, 0.);
+            for (size_t i = 0; i < extents_element; ++i) {
+              for (size_t j = 0; j < extents_mortar; ++j) {
+                for (size_t k = 0; k < extents_element; ++k) {
+                  projection(i, j) += spectral_to_grid_element(i, k) *
+                                      grid_to_spectral_mortar(k, j);
+                }
+              }
+            }
+
+            return projection;
+          });
+      return cache(encode_quadrature(element_mesh.quadrature(0)),
+                   element_mesh.extents(0),
+                   encode_quadrature(mortar_mesh.quadrature(0)),
+                   mortar_mesh.extents(0));
+    }
+
+    case MortarSize::UpperHalf: {
+      const static StaticCache<
+          Matrix, CacheRange<0, supported_quadratures.size()>,
+          CacheRange<2, maximum_number_of_points<Basis::Legendre> + 1>,
+          CacheRange<0, supported_quadratures.size()>,
+          CacheRange<2, maximum_number_of_points<Basis::Legendre> + 1>>
+          cache([](const size_t encoded_quadrature_element,
+                   const size_t extents_element,
+                   const size_t encoded_quadrature_mortar,
+                   const size_t extents_mortar) noexcept {
+            if (extents_element > extents_mortar) {
+              return Matrix{};
+            }
+            const Mesh<1> mesh_element(
+                extents_element, Basis::Legendre,
+                decode_quadrature(encoded_quadrature_element));
+            const Mesh<1> mesh_mortar(
+                extents_mortar, Basis::Legendre,
+                decode_quadrature(encoded_quadrature_mortar));
+
+            // The transformation from the small interval to the large
+            // interval in spectral space.  This is a rearranged
+            // version of the equation given in the header.
```

Comment:
I'm not sure what to say.  I rearranged them because I like doing math and I found a rearranged form that was easier to code, but that's not a very useful comment.  The comment is there as a warning to anyone who tries to compare the code with the docs.

Reply:
Maybe just mention the rearranged form is easier to code? :) That'll satisfy anyone wondering why they're different

Reply:
Sounds good.